[IMP] l10n_tr_nilvera_einvoice: check invoice name prior XML generation

ijja-odoo opened the pull request:
Before this commit:
- When sending an e-invoice through Nilvera, invoice name was not validated, which could raise errors during XML generation.

After this commit:
- Invoice name is validated before sending the e-invoice through Nilvera to ensure they comply with the GIB format, to prevent XML generation errors.

task-5117359

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/229270.png)](https://mergebot.odoo.com/odoo/odoo/pull/229270)

ijja-odoo commented:
@gawa-odoo @SaddemAmine Requested changes have been made, can you please have a look?

gawa-odoo commented on addons/l10n_tr_nilvera_einvoice/models/account_move_send.py:
```diff
@@ -24,6 +25,16 @@ def _get_all_extra_edis(self) -> dict:
+                parts['year'] != 0
+                and parts['year_length'] == 4
+                and parts['seq'] != 0
+                and re.match(r'^[A-Za-z0-9]{3}[^A-Za-z0-9]?$', parts['prefix1'])
```
is teh regex necessary? 
Can't you just check isalnum and len==3? 
Wouldn't it be cleaner?

ijja-odoo commented:
> Why are you changing the tests? You don't seem to enter it anyway, right?
* I changed the test because the existing expected XMLs were based on the `.split('/')` logic, and the einvoice name used in the test was invalid. (see: https://runbot.odoo.com/runbot/build/91263848).

> Else your sequence at 0 would fail, no?
* The name validation happens inside `_get_alerts()`, so it doesn’t triggers during XML generation.

> I think you should add a quick test for it.
* I’ve added the tests. Thanks!
